### DISCUSSION TITLE
What is the difference between `templates/cloudflare-workers` and `templates/cloudflare`

### DISCUSSION THREAD

**Discussion by flexchar** [remix-run/remix#9641] [Q&A] [answered]
What is the difference between `templates/cloudflare-workers` and `templates/cloudflare`

**Comment by Master-Hash** [comment, marked as answer]
The latter use Cloudflare Pages, which is the recommended way to deploy full stack web applications. 

Workers and Pages are almost the same, but their usages are different to fit different jobs, e.g. Pages have CI integration, static files uploading, etc, but cannot use Cron jobs or email handler.

Most functions, like KV, R2, D1, work in both products.

It may change, because Cloudflare is unifying these two products.

https://blog.cloudflare.com/pages-and-workers-are-converging-into-one-experience

May this helps you!

**Reply by flexchar** [reply]
thank you Hash! this is really helpful. It would be lovely to have this as a part of templates page in the docs. 

https://remix.run/docs/en/main/guides/templates#official-templates
The ambiguity in the path makes it difficult to decide which one I should go with. :)

**Reply by flexchar** [reply]
The Cloudflare article is almost a year old. I wonder if you have any personal input to which is better excluding the certain unavailable features. Perhaps the performance differs, one being faster another focusing on offering more rich feature set.

**Reply by Master-Hash** [reply]
https://github.com/remix-run/remix/discussions/1248 Here's an old discussion 3 years ago, and still relevant.

I thought Pages would be fine if you have no idea.
